onnxscript/function_libs/torch_lib/graph_building.py

```diff
@@ -368,13 +368,22 @@ def eval_function(  # type: ignore[override]
         ) = param_manipulation.separate_input_attributes_from_arguments(
             param_schemas, args, kwargs, fill_defaults=True, allow_extra_kwargs=True
         )
-        name_to_schema = {param.name: param for param in param_schemas}
+
+        # Cast attributes to the correct type based on function signature
+        op_schema = function.op_schema
+        assert op_schema is not None
         for name, value in attributes.items():
-            param = name_to_schema[name]
+            attribute = op_schema.attributes[name]
             # Cast int to float if needed
-            if param.type in {float, "float"}:
-                # FIXME(justinchuby): Create invariant on the type of param.type to simplify this
+            if attribute.type == onnx.defs.OpSchema.AttrType.FLOAT:
+                # Cast int to float is the attribute is FLOAT
                 attributes[name] = float(value)
+            if attribute.type == onnx.defs.OpSchema.AttrType.INTS and isinstance(value, int):
+                attributes[name] = (value,)
```

Comment:
when would this happen?

Reply:
In PyTorch, attributes annotated as `int[1]?` accepts an integer or a sequence. When the attribute is an integer, it is treated as a single element sequence. ONNX requires an attribute to either be an integer or a sequence. So we promote the value to a sequence here.

Added as a comment too.